### PR TITLE
Simplify assembly

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
Using go:linkname and a call5 trampoline function it is possible to remove assembly and push it into Go code. Arguably, it is pretty ugly Go code but functional. Having less assembly means less chance to make a copy paste error and make it easier to port to other architectures

### CODE REVIEW

File: internal/fakecgo/trampolines_stubs.s [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+//go:build darwin
+// +build darwin
+
+#include "textflag.h"
+
+// these stubs are here because it is not possible to go:linkname directly the C functions on darwin arm64
```
Comment (hajimehoshi) [inline internal/fakecgo/trampolines_stubs.s:9]: Is this file needed only for darwin/arm64?

Comment (TotallyGamerJet) [reply internal/fakecgo/trampolines_stubs.s:9]: No they are needed on both amd64 and arm64. The reason it has to be done this way is bc arm64 doesn't allow directly linking to dynamic C functions

File: internal/fakecgo/trampolines_stubs.s [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+//go:build darwin
+// +build darwin
```
Comment (hajimehoshi) [inline internal/fakecgo/trampolines_stubs.s:5]: Let's rename the file to `trampolines_stubs_darwin.s` or `trampolines_darwin.s` instead of build tags here

**Review by hajimehoshi — APPROVED** [review]
LGTM

**Comment by hajimehoshi** [comment]
Oh sorry I should have let you merge this as usual...

**Comment by TotallyGamerJet** [comment]
All good! It happens
